### PR TITLE
Feature/chat wallet app extraction

### PR DESCRIPTION
**Pull request by alexander-lsvk**
- Remove wallet from **Chat** app
- Implement new **Wallet** app with updated design using **Web3Wallet** lib

* [ ] Breaking change
* [x] Requires a documentation update

### CODE REVIEW

File: Example/WalletApp/ApplicationLayer/SceneDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          return sceneConfig
+      }
+
+    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+        guard let windowScene = (scene as? UIWindowScene) else { return }
+
+        window = UIWindow(windowScene: windowScene)
+        window?.makeKeyAndVisible()
+        
+        app.uri = connectionOptions.urlContexts.first?.url.absoluteString.replacingOccurrences(of: "walletapp://wc?uri=", with: "")
```
Comment (llbartekll) [inline Example/WalletApp/ApplicationLayer/SceneDelegate.swift:31]: we need to switch to universal linking later

File: Example/WalletApp/DomainLayer/AccountStorage/AccountStorage.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import Foundation
+
+final class AccountStorage {
```
Comment (llbartekll) [inline Example/WalletApp/DomainLayer/AccountStorage/AccountStorage.swift:3]: I cannot find any usage of this class in the WalletApp

Comment (llbartekll) [reply Example/WalletApp/DomainLayer/AccountStorage/AccountStorage.swift:3]: Ok it's in Application but never used after. Do we need it?

File: Example/WalletApp/ApplicationLayer/Application.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import Foundation
+import WalletConnectChat
+
+final class Application {
+    var uri: String?
+    
+    lazy var chatService: ChatService = {
```
Comment (llbartekll) [inline Example/WalletApp/ApplicationLayer/Application.swift:7]: do we need chat in WalletApp?

File: Example/WalletApp/DomainLayer/Chat/AccountNameResolver.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+struct AccountNameResolver {
```
Comment (llbartekll) [inline Example/WalletApp/DomainLayer/Chat/AccountNameResolver.swift:3]: same here, it's never used by WalletApp

File: Example/WalletApp/DomainLayer/Chat/ChatService.swift [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import Foundation
+import Combine
+import WalletConnectChat
+import WalletConnectRelay
+
+typealias Stream<T> = AsyncPublisher<AnyPublisher<T, Never>>
+
+final class ChatService {
```
Comment (llbartekll) [inline Example/WalletApp/DomainLayer/Chat/ChatService.swift:8]: same,
I think if we ever integrate chat into walletApp it will be done with web3inbox

File: Example/WalletApp/DomainLayer/Chat/RegisterService.swift [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+final class RegisterService {
```
Comment (llbartekll) [inline Example/WalletApp/DomainLayer/Chat/RegisterService.swift:3]: same

File: Sources/Web3Wallet/Web3WalletClient.swift [hunk 7]
```diff
@@ -29,6 +29,13 @@ public class Web3WalletClient {
     public var authRequestPublisher: AnyPublisher<AuthRequest, Never> {
         authClient.authRequestPublisher.eraseToAnyPublisher()
     }
+    
+    /// Publisher that sends sessions on every sessions update
+    ///
+    /// Event will be emited on controller and non-controller clients.
+    public var sessionsPublisher: AnyPublisher<[Session], Never> {
```
Comment (llbartekll) [inline Sources/Web3Wallet/Web3WalletClient.swift:36]: 👍

**Review by llbartekll — COMMENTED** [review]
I build the app looks really nice, just the session proposal is not appearing but we've covered that on meeting yesterday so waiting for update.

**Comment by alexander-lsvk** [comment]
@llbartekll updated with session request and PR fixes

**Review by flypaper0 — COMMENTED** [review]
Showcase app crash on "Start messaging" press

**Comment by flypaper0** [comment]
Is it possible to fix modal background appearance on "Enter a WalletConnect URI" screen? Not critical

**Comment by flypaper0** [comment]
Why WalletApp redirect to Showcase after Approving Auth Request?

File: Example/Showcase/Classes/DomainLayer/SocketFactory/SocketFactory.swift [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+import Starscream
+import WalletConnectRelay
+
+struct SocketFactory: WebSocketFactory {
```
Comment (flypaper0) [inline Example/Showcase/Classes/DomainLayer/SocketFactory/SocketFactory.swift:5]: You can use DefaultSocketFactory from Shared folder. Just include it in WalletApp target

File: Example/Showcase/Classes/PresentationLayer/Chat/Welcome/WelcomeInteractor.swift [hunk 9]
```diff
@@ -1,9 +1,17 @@
...
+import WalletConnectPairing
+import Auth
 
 final class WelcomeInteractor {
+    private var disposeBag = Set<AnyCancellable>()
+    
     private let chatService: ChatService
     private let accountStorage: AccountStorage
+    
+    var authClient: AuthClient?
```
Comment (flypaper0) [inline Example/Showcase/Classes/PresentationLayer/Chat/Welcome/WelcomeInteractor.swift:14]: Why authClient is setting in generateUri? If important to initialize it lazily it could be lazy var. 
Didn't find, why we are store it?

File: Example/WalletApp/ApplicationLayer/Application.swift [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import Foundation
+import WalletConnectChat
+
+final class Application {
+    var uri: String?
```
Comment (flypaper0) [inline Example/WalletApp/ApplicationLayer/Application.swift:5]: where it's used?

File: Example/WalletApp/ApplicationLayer/SceneDelegate.swift (hunk 1 above)
Comment (flypaper0) [inline Example/WalletApp/ApplicationLayer/SceneDelegate.swift:31]: What the idea of doing that?

File: Example/WalletApp/Common/Extensions/UIKit/UIViewController.swift [hunk 11]
```diff
@@ -0,0 +1,59 @@
+import UIKit
+import SwiftUI
+import StoreKit
+
+extension UIViewController {
```
Comment (flypaper0) [inline Example/WalletApp/Common/Extensions/UIKit/UIViewController.swift:5]: We need to share common things between our apps with SPM Package. Could be a good improvements may be in another PR's

File: Example/WalletApp/Common/Extensions/UIKit/UIViewController.swift [hunk 12]
```diff
@@ -0,0 +1,59 @@
+import UIKit
+import SwiftUI
+import StoreKit
```
Comment (flypaper0) [inline Example/WalletApp/Common/Extensions/UIKit/UIViewController.swift:3]: StoreKit?

File: Example/WalletApp/DomainLayer/SocketFactory/SocketFactory.swift [hunk 13]
```diff
@@ -0,0 +1,11 @@
+import Foundation
+import Starscream
+import WalletConnectRelay
+
+extension WebSocket: WebSocketConnecting { }
+
+struct SocketFactory: WebSocketFactory {
```
Comment (flypaper0) [inline Example/WalletApp/DomainLayer/SocketFactory/SocketFactory.swift:7]: same as https://github.com/WalletConnect/WalletConnectSwiftV2/pull/648/files#r1063438411

**Comment by alexander-lsvk** [comment]
> Why WalletApp redirect to Showcase after Approving Auth Request?

@flypaper0 in showcase you receive Cacao object after, this redirection is kind of success imitation, but I can remove it or change if needed

File: Example/WalletApp/PresentationLayer/Wallet/ConnectionDetails/ConnectionDetailsView.swift [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+                                            .foregroundColor(.grey50)
+                                            .font(.system(size: 13, weight: .semibold, design: .rounded))
+                                        
+                                        Spacer()
+                                    }
+                                    .padding(.horizontal, 18)
+                                    .padding(.top, 10)
+                                    
+                                    HStack(spacing: 0) {
+                                        Text("\(presenter.session.namespaces[namespace]?.accounts.first?.namespace ?? "") : \((presenter.session.namespaces[namespace]?.accounts.first?.reference ?? ""))")
```
Comment (flypaper0) [inline Example/WalletApp/PresentationLayer/Wallet/ConnectionDetails/ConnectionDetailsView.swift:63]: I thing that kind of complex things is better put into presenter: `presenter.namespace`

File: Example/WalletApp/ApplicationLayer/SceneDelegate.swift (hunk 1 above)
Comment (alexander-lsvk) [reply Example/WalletApp/ApplicationLayer/SceneDelegate.swift:31]: The logic behind is to get the **wc uri** if app was opened via deeplink and to be able to handle within any screen in the wallet app.
Like in the current UI we have a welcome screen which is not responsible for setting any kind of subscriptions and performing some requests.
So when I save **uri** to the **app**, I can trigger handling in the next screen - **WalletView** or any other.

File: Example/WalletApp/PresentationLayer/Wallet/PasteUri/PasteUriView.swift [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+                .edgesIgnoringSafeArea(.all)
+            
+            VStack {
+                Spacer()
+                VStack(spacing: 6) {
+                    Text("Enter a WalletConnect URI")
+                        .foregroundColor(.grey8)
+                        .font(.system(size: 22, weight: .bold, design: .rounded))
+                    
+                    Text("To get the URI press the 􀐅 copy to clipboard button in wallet connection interfaces.")
```
Comment (flypaper0) [inline Example/WalletApp/PresentationLayer/Wallet/PasteUri/PasteUriView.swift:26]: mb remove this signs? "􀐅"

File: Example/WalletApp/PresentationLayer/Wallet/Scan/ScanView.swift [hunk 16]
```diff
@@ -0,0 +1,48 @@
+import SwiftUI
+
+struct ScanView: View {
+    @EnvironmentObject var presenter: ScanPresenter
+
+    var body: some View {
+        ZStack {
+            //        ScanQR(onValue: presenter.onValue, onError: presenter.onError)
```
Comment (flypaper0) [inline Example/WalletApp/PresentationLayer/Wallet/Scan/ScanView.swift:8]: comments

**Review by flypaper0 — APPROVED** [review]
